e2e/e2e_test.go

```diff
@@ -0,0 +1,301 @@
+				pod, err := cs.CoreV1().Pods(corev1.NamespaceDefault).Get(appName, metav1.GetOptions{})
+				if err != nil {
+					if kerrors.IsNotFound(err) {
+						return false, nil
+					}
+					return false, err
+				}
+				appPod = pod
+				for _, cond := range appPod.Status.Conditions {
+					if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+				return false, nil
+			}); err != nil {
+				t.Fatalf("failed to observe ready example pod %q in time: %s\npod: %v", appName, err, appPod)
+			}
+			l.Printf("Took %v for pod to become ready\n", time.Since(start))
+
+			// Create service object.
+			svcName := "svc"
+			svc := corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: svcName,
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"app": appName,
+					},
+					Type: corev1.ServiceTypeLoadBalancer,
+					Ports: []corev1.ServicePort{
+						{
+							Port: 80,
+						},
+					},
+				},
+			}
+
+			if _, err := cs.CoreV1().Services(corev1.NamespaceDefault).Create(&svc); err != nil {
+				t.Fatalf("failed to create service: %s", err)
+			}
+			// External LBs don't seem to get deleted when the kops cluster is
+			// removed, at least not on DO. Hence, we'll do it explicitly.
+			defer func() {
+				if err := cs.CoreV1().Services(corev1.NamespaceDefault).Delete(svcName, &metav1.DeleteOptions{}); err != nil {
+					t.Fatalf("failed to delete service: %s", err)
+				}
+			}()
+
+			// Wait for service IP address to be assigned.
+			l.Println("Polling for service load balancer IP address assignment")
+			start = time.Now()
+			var lbAddr string
+			if err := wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
+				svc, err := cs.CoreV1().Services(corev1.NamespaceDefault).Get(svcName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				for _, ing := range svc.Status.LoadBalancer.Ingress {
+					lbAddr = ing.IP
+					return true, nil
+				}
+
+				return false, nil
+			}); err != nil {
+				t.Fatalf("failed to observe load balancer IP address assignment: %s", err)
+			}
+			l.Printf("Took %v for load balancer to get its IP address assigned\n", time.Since(start))
+
+			// Send request to the pod over the LB.
+			cl := &http.Client{
+				Timeout: 5 * time.Second,
+			}
+			u := fmt.Sprintf("http://%s:80", lbAddr)
+
+			var attempts, lastStatusCode int
+			if err := wait.Poll(1*time.Second, 3*time.Minute, func() (bool, error) {
+				l.Printf("Sending request to %s", u)
+				attempts++
+				resp, err := cl.Get(u)
+				if err != nil {
+					return false, nil
+				}
+				defer resp.Body.Close()
+
+				lastStatusCode = resp.StatusCode
+				if resp.StatusCode != http.StatusOK {
+					return false, nil
+				}
+
+				return true, nil
+			}); err != nil {
+				t.Fatalf("failed to send request over LB to example application: %s (last status code: %d / attempts: %d)", err, lastStatusCode, attempts)
```

Comment:
Prefer t.Errorf since it's a test case failure not a setup failure.

Reply:
Same as above. :)